### DISCUSSION TITLE
Simpler way to exclude fields on the fly from the log?

### DISCUSSION THREAD

**Discussion by wrabit** [spatie/laravel-activitylog#844] [Q&A]
I'd like to exclude specific fields from being logged, so in my models I can have something like:

```
protected static $excludeFromLog = ['owner_id'];
```

and in a trait
```
    public function tapActivity(Activity $activity, string $eventName)
    {
        if (property_exists(get_class($this), 'excludeFromLog')) {
            foreach (self::$excludeFromLog as $exc) {
                $properties = $activity->properties->toArray();
                \Arr::forget($properties, 'attributes.'.$exc);
                \Arr::forget($properties, 'old.'.$exc);
                $activity->properties = $properties;
            }
        }
    }
```

The field in this example is not excluded by guarded.

**Comment by Gummibeer** [comment]
Hey,
we already provide a lot of possibilities to customize the logged attributes.
https://spatie.be/docs/laravel-activitylog/v3/advanced-usage/logging-model-events
It's too complex for most users and even for us in some cases.
Why do you have the attribute in your logged attributes in the first place if you want to exclude them?

**Reply by wrabit** [reply]
I have a series of forms (models) some of which have many fields (70 plus). Instead of itemising each field in a fillable, I use guarded. Guarded is declared in a parent model and occasionally some local field is also to be ignored, which can't be added to the parent's guarded. Its an api and I use resources which successfully weeds out the sensitive fields from the front.
